src/mozanalysis/frequentist_stats/sample_size.py

```diff
@@ -391,7 +398,86 @@ def _get_sample_size_col(col):
         pop_percent = 100.0 * (sample_size / len(df))
         results[col] = {
             "sample_size_per_branch": sample_size,
-            "number_of_clients_targeted": len(df),
             "population_percent_per_branch": pop_percent,
+            "number_of_clients_targeted": len(df),
         }
     return results
+
+
+def variable_enrollment_length_sample_size_calc(
+    bq_context: BigQueryContext,
+    start_date: Union[str, datetime],
+    max_enrollment_days: int,
+    analysis_length: int,
+    metric_list: List[Metric],
+    target_list: List[Segment],
+    variable_window_length: int = 7,
+    experiment_name: str = "",
+    app_id: Optional[str] = "",
+    to_pandas: bool = True,
+    **sizing_kwargs,
+):
+    if variable_window_length > max_enrollment_days:
+        raise ValueError(
+            "Enrollment window length is larger than the max enrollment length."
+        )
+
+    ht = HistoricalTarget(
+        start_date=start_date,
+        analysis_length=analysis_length,
+        num_dates_enrollment=max_enrollment_days,
+        experiment_name=experiment_name,
+        app_id=app_id,
+    )
+
+    df = ht.get_single_window_data(
+        bq_context=bq_context, metric_list=metric_list, target_list=target_list
+    )
+
+    def _get_time_intervals(start_date: Union[str, datetime], interval: int):
```

Comment:
I'm a fan of defining functions at the top level (i.e., not nested within other functions) because that way they can be independently unit tested. Especially for a simple function like this that doesn't need any extra state.

Reply:
Also, can we add type hinting for the return type? From reading the code I see it's returning a `Dict[int, datetime]`, but that would be nice in the signature. Also maybe a doc string about what it means? What does the int key mean here?

Reply:
Yeah, that makes sense. I'll move this function into util.py and add type hinting for the return.